src/pages/home/report/EmojiPickerMenu.js

```diff
@@ -0,0 +1,111 @@
+import emojis from '../../../../assets/emojis';
+import EmojiPickerMenuItem from './EmojiPickerMenuItem';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+
+const propTypes = {
+    // Controls the visibility of this component.
+    isVisible: PropTypes.bool,
+    addEmojiToTextBox: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    isVisible: false,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 500, false);
+        this.state = {
+            filteredEmojis: emojis,
+            headerIndices: [0, 34, 60, 88, 99, 121, 148],
+        };
+    }
+
+    componentDidUpdate() {
+        if (this.props.isVisible) {
+            this.searchInput.focus();
+        }
+    }
+
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            this.setState({filteredEmojis: emojis, headerIndices: [0, 34, 60, 88, 99, 121, 148]});
+            return;
+        }
+        const newFilteredEmojiList = [];
+        emojis.forEach((emoji) => {
+            if (!emoji.header && emoji.code !== 'BLANK') {
+                emoji.keywords.forEach((keyword) => {
+                    if (keyword.includes(normalizedSearchTerm)) {
+                        newFilteredEmojiList.push(emoji);
+                    }
+                });
+            }
+        });
+
+        this.setState({filteredEmojis: newFilteredEmojiList, headerIndices: []});
+    }
+
+    renderItem({item}, addEmojiToTextBox) {
+        if (item.code === 'BLANK') {
+            return;
+        }
+
+        if (item.header) {
+            return (
+                <Text style={styles.emojiHeaderStyle}>
+                    {item.code}
+                </Text>
+            );
+        }
+
+        return (
+            <EmojiPickerMenuItem
+                onPress={addEmojiToTextBox}
+                emoji={item.code}
+            />
+        );
+    }
+
+    render() {
+        return (
+            this.props.isVisible && (
+                <View style={[styles.emojiPickerContainer]}>
+                    <View style={[styles.pt4, styles.ph4, styles.pb1]}>
+                        <TextInputFocusable
+                            textAlignVertical="top"
+                            placeholder="Search"
+                            placeholderTextColor={themeColors.textSupporting}
+                            onChangeText={this.filterEmojis}
+                            style={[styles.textInput]}
+                            defaultValue=""
+                            ref={el => this.searchInput = el}
+                        />
+                    </View>
+                    <FlatList
+                        data={this.state.filteredEmojis}
+                        renderItem={item => this.renderItem(item, this.props.addEmojiToTextBox)}
```

Comment:
So just FYI, by using an arrow function for `renderItem`, it's re-declared as a new function each time this component is rendered, so `FlatList` sees it as a changed prop, and in turn the `FlatList` will re-render, which is bad for performance.

So what I think we should probably do instead is just have `renderItem` use `this.props.addEmojiToTextBox` internally and bind it to this component. Does that work?